### ISSUE TITLE
Fast parser should warn about using ellipses with other function arguments

### ISSUE THREAD

**Issue by ddfisher** [python/mypy#2704]
For this function: 
```python
def f(x, y, z): # type: (..., int) -> None
     pass
```
The old parser would give the warning: `Ellipses cannot accompany other argument types in function type signature.`, whereas the fast parser gives the less helpful `Type signature has too few arguments`.

The fast parser should look for ellipses args and give a special-cased better error message.
